pkg/platform/local/platform.go

```diff
@@ -83,60 +83,76 @@ func (p *Platform) DeployFunction(deployOptions *platform.DeployOptions) (*platf
 
 // GetFunctions will return deployed functions
 func (p *Platform) GetFunctions(getOptions *platform.GetOptions) ([]platform.Function, error) {
-	getContainerOptions := &dockerclient.GetContainerOptions{
-		Labels: map[string]string{
-			"nuclio-platform":  "local",
-			"nuclio-namespace": getOptions.Namespace,
-		},
-	}
-
-	// if we need to get only one function, specify its function name
-	if getOptions.Name != "" {
-		getContainerOptions.Labels["nuclio-function-name"] = getOptions.Name
-	}
-
-	containersInfo, err := p.dockerClient.GetContainers(getContainerOptions)
-
-	if err != nil {
-		return nil, errors.Wrap(err, "Failed to get containers")
+	var getContainerOptionsSlice []*dockerclient.GetContainerOptions
+
+	// initialize getContainerOptionsSlice with same number of items like getOptions.MatchCriterias
+	for a := 0; a < len(getOptions.MatchCriterias); a++ {
```

Comment:
You should honestly just use one `range` over `getOptions.MatchCriterias` where you're both:
- creating a new `dockerclient.GetContainerOptions`
- putting the `nuclio-platform` and `nuclio-namespace` values in its Labels
- (if matchCriteria has a Name) setting the `nuclio-function-name` label too
- appending it to the slice

(also never use `a`)